adam-core/src/main/scala/org/bdgenomics/adam/rdd/GenomicPartitioners.scala

```diff
@@ -89,6 +89,16 @@ case class GenomicPositionPartitioner(numParts: Int, seqLengths: Map[String, Lon
         )
         getPart(refpos.referenceName, refpos.pos)
       }
+      case (refpos: ReferencePosition, k: Any) => {
+        require(
```

Comment:
There has got to be a cleaner way to do this. How can I piggyback these cases?
case (refpos: ReferencePosition) | (refpos: ReferencePosition, k: Any) throws an error because refpos cannot be defined twice...